elide-core/src/main/java/com/yahoo/elide/core/QueryDetail.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Value;
+
+/**
+ * QueryDetail implementation class. Stores the model name (root entity class) and
+ * the queryText that the underlying datastore will eventually run
+ */
+@Value
+@AllArgsConstructor
+@NoArgsConstructor(force = true)
+public class QueryDetail {
+    @Getter private String modelName;
```

Comment:
Instead of modelName - what if we changed this to the query itself (of type Query).

Reply:
On second thought, It might be cleaner if we just made these arguments to the processQuery function.  QueryDetail is missing some fields (like the requestId) - which makes me think they all either need to come together - or we should keep them as separate arguments.